core/server/src/main/java/tachyon/master/file/FileSystemMasterRestServiceHandler.java

```diff
@@ -0,0 +1,392 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.master.file;
+
+import java.io.IOException;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import tachyon.Constants;
+import tachyon.TachyonURI;
+import tachyon.exception.TachyonException;
+import tachyon.master.MasterContext;
+import tachyon.master.TachyonMaster;
+import tachyon.master.file.options.CompleteFileOptions;
+import tachyon.master.file.options.CreateDirectoryOptions;
+import tachyon.master.file.options.SetAclOptions;
+import tachyon.master.file.options.SetAttributeOptions;
+
+/**
+ * This class is a REST handler for file system master requests.
+ */
+@Path("/")
+public final class FileSystemMasterRestServiceHandler {
```

Comment:
it will have same thread-safe guarantee as other FileSystem service handler ... in particular, it is currently does not meet the thread-safety definition, but things seem to be working ;-)